src/odemis/gui/cont/acquisition.py

```diff
@@ -1396,10 +1396,13 @@ def __init__(self, tab_data, tab_panel, projectbar_ctrl, calibration_ctrl):
         self._calibration_ctrl = calibration_ctrl
 
         # Path to the acquisition
-        # <External storage>/<date>/<project name>/<roa name>
-        # TODO: display proper path to external storage (if mounted)
+        # <Path to External storage>/<date>/<project name>/<roa name>
         self.path = datetime.today().strftime('%Y-%m-%d')
-        self._tab_panel.txt_destination.SetValue("<External Storage>/%s" % self.path)
+        if self._main_data_model.asm:
+            self._main_data_model.asm.externalStorageURL.subscribe(self._on_storage_dir, init=True)
+        else:
+            # Should never happen on a real system, however, it's useful for the simulators
```

Comment:
This is not completely true. The ASM simulator will return a path. Only the simulators without the asm component will need this hack, which at the moment is the one in usr/bin/share/odemis/sim. So I suggest to say, 
`# Should never happen on a real system, however, it's useful for simulators without an asm component`

Reply:
done